### PR TITLE
Upgraded to electron 7.x.x

### PR DESCRIPTION
**Pull request by EricSmekens**
Decided to upgrade to electron 7.x.x

- Had to update some methods that were moved to promises instead of callbacks.
- Had to pass in the `getCurrentWindow` to make the dialogs modal again.

Ran some tests, all was looking good.

Install size for windows decreased significantly with Electron 6/7.x.x.
And must be running slightly faster as well due to Chromium and Nove/V8 improvements.

### CODE REVIEW

**Comment by fenos** [comment]
@EricSmekens This is awesome!!! 

I'll give it a play around and then merge it! 

Thanks!

**Comment by EricSmekens** [comment]
Hi @fenos,

Did you have time to look into this or other issues?
If you need somebody to spend some time on that/help collaborating, I'm willing to as well.

File: package.json [hunk 1]
```diff
@@ -163,9 +163,9 @@
     "css-loader": "^2.1.1",
     "detect-port": "^1.3.0",
     "devtron": "^1.4.0",
-    "electron": "6.1.7",
-    "electron-builder": "^20.38.5",
-    "electron-debug": "^2.1.0",
+    "electron": "7.1.11",
```
Comment (oneslash) [inline package.json:166]: do you think you can upgrade to 7.1.14 or to the latest version of 7x branch?

Comment (EricSmekens) [reply package.json:166]: Yes, I don't see why not.
(Tested with 7.1.11, wish to upgrade to version 9 if this branch gets merged).

**Comment by EricSmekens** [comment]
Would resolving the conflict and update to 7.1.14 help to get this PR merged or not?
Would be able to spend time on that, if that is the case.

**Comment by EricSmekens** [comment]
@fenos Would you be interested in a pull request to the latest version of electron instead?

**Review by jackielii — APPROVED** [review]
LGTM thanks

**Review by jackielii — APPROVED** [review]

**Pull request merged**